Try everything to abort backups on metered network (when not allowed)

grote opened the pull request:
This does several improvements to not cause traffic while user is on mobile data:

* Halt app backup worker progress
* Improve app backup error notification: add try again button and language when error was due to metered network
* stops apk backup
* show error dialog when manually requesting backup
* stop app backups
* stop file backups

This also treats `SocketException` as transient.

Fixes #794

grote committed:
Halt AppBackupWorker progress if we get stopped or entered metered network

grote committed:
Treat SocketException as a transient exception in WebDavBackend

grote committed:
Improve app backup error notification

Add 'Try Again' button and show different text for metered network

grote committed:
Don't continue with APK backup if we move to metered network

grote committed:
Lazily get info in BackupRequester

so it only requests info when actually needed

grote committed:
Show dialog when starting backup on metered network

grote committed:
Abort app backups when we switched to metered network

We don't request new package chunks for backup and reject backup for all packages already requested.

grote committed:
Halt file backup when switched to metered network

grote committed:
Adapt metered network strings to feedback

t-m-w commented:
Apps backup behavior appears to work fine when switching to a metered network: backup stops, and I get the notification. Files backup, however, does not currently appear to halt as expected. I shared some logs privately.

grote committed:
Also abort file backup while backup up a single huge file

previously switching to a metered network would have left the upload to continue until the file would be complete

grote commented:
There was an issue that backup wasn't aborted while uploading a single huge file. Latest commit should address this.

t-m-w commented:
> There was an issue that backup wasn't aborted while uploading a single huge file. Latest commit should address this.

It's working now! I'm going to approve from my side but have a couple small UX notes:
1. When files backup is interrupted, there is no notification to try again like there is for other backup.
2. The "Try again" notification action dismisses the notification without doing anything, if you are still on a metered network when you use it.
3. I have not tested what happens for automatic backups and whether it's different from manual backups.

t-m-w reviewed (approved):
Seems good to me! See previous comment for a couple small notes.

grote commented:
> When files backup is interrupted, there is no notification to try again like there is for other backup.

That's because there's no error notification for file backup at all, yet. There's probably a ticket for that.

> The "Try again" notification action dismisses the notification without doing anything, if you are still on a metered network when you use it.

Hmm.. true this could be improved, but I agree not a deal breaker for the feature to land.

> I have not tested what happens for automatic backups and whether it's different from manual backups.

Let's hope it works the same :D
